djstripe/stripe_objects.py

```diff
@@ -1199,10 +1199,13 @@ class Meta:
     def api_retrieve(self, api_key=settings.STRIPE_SECRET_KEY):
         # OVERRIDING the parent version of this function
         # Cards must be manipulated through a customer or account.
-        # TODO: When managed accounts are supported, this method needs to check if either a customer or
-        #       account is supplied to determine the correct object to use.
-
-        return self.customer.api_retrieve(api_key=api_key).sources.retrieve(self.stripe_id, expand=self.expand_fields)
+        # TODO: When managed accounts are supported, this method needs to check if
+        # either a customer or account is supplied to determine the correct object to use.
+        customer = self.customer.api_retrieve(api_key=api_key)
+        if "sources" not in customer:
```

Comment:
[style] Quick newlines before the conditional and before the return and it's good to do.